File: hello.py
# HOWTO:
#
# Python program files are named SOMETHING.scm.
#
# Run w/ 0 arguments:
# $ python3 hello.py
#
# Run w/ 1 argument:
# $ python3 hello.py Tau

import sys

def length(blah):
    return len(blah)

def main(words):
    if length(words) == 2:     # why 2?
        print("Hello, " + words[1] + "!")
    else:
        print("Hello, STRANGER!")
    sys.exit(0)

main(sys.argv)

File: hello.c
// HOWTO:
//
// C program files are named SOMETHING.c.
// A "compiler" named "cc" compiles SOMETHING.c into SOMETHING.exe.
//
// Compile to create program ./hello.exe:
// $ cc hello.c -o ./hello.exe
//
// Run w/ 0 arguments:
// $ ./hello.exe
//
// Run w/ 1 argument:
// $ ./hello.exe Tau

#include <stdio.h>

int main(int number_of_words, char** words) {
  if ( number_of_words == 2 ) {      // why 2?
    printf("Hello, %s!\n", words[1]);
  } else {
    printf("Hello, %s!\n", "STRANGER");
  }
  return 0;}
